File: programmingModels/OpenMP/demo/01_target_construct.cpp
#include <stdio.h>
#include <omp.h>

int main( int argv, char** argc ) {

  printf( "Number of devices: %d\n", omp_get_num_devices() );

  #pragma omp target
  {
    if( !omp_is_initial_device() )
      printf( "Hello world from accelerator.\n" );
    else
      printf( "Hello world from host.\n" );
  }
  return 0;
}

File: programmingModels/OpenMP/demo/fortran/01_target_construct.f90
program  main
  use omp_lib
  implicit none

  print *, "Number of devices:", omp_get_num_devices()

  !$omp target
   if( .not. omp_is_initial_device() ) then
     print *, "Hello world from accelerator"
   else
     print *, "Hello world from host"
  endif
  !$omp end target

end program main
